intel_extension_for_transformers/neural_chat/tools/embedding_finetune/README.md

```diff
@@ -0,0 +1,139 @@
+--passage_max_len 256 \
+--train_group_size 2 \
+--logging_steps 10 \
+--query_instruction_for_retrieval "" \
+--bf16 True
+```
+
+**Some Important Arguments**:
+- `per_device_train_batch_size`: batch size in training. In most cases, a larger batch size will bring stronger performance. 
+- `train_group_size`: the number of positives and negatives for a query in training.
+There is always one positive, so this argument will control the number of negatives (#negatives=train_group_size-1).
+Noted that the number of negatives should not be larger than the number of negatives in data `"neg": List[str]`.
+Besides the negatives in this group, the in-batch negatives also will be used in fine-tuning.
+- `negatives_cross_device`: share the negatives across all GPUs. This argument will extend the number of negatives.
+- `learning_rate`: select an appropriate for your model. Recommend 1e-5/2e-5/3e-5 for large/base/small-scale. 
+- `temperature`: It will influence the distribution of similarity scores.
+- `query_max_len`: max length for query. Please set it according to the average length of queries in your data.
+- `passage_max_len`: max length for passage. Please set it according to the average length of passages in your data.
+- `query_instruction_for_retrieval`: instruction for query, which will be added to each query. You also can set it `""` to add nothing to the query.
+- `use_inbatch_neg`: use passages in the same batch as negatives. The default value is True. 
+
+For more training arguments please refer to [transformers.TrainingArguments](https://huggingface.co/docs/transformers/main_classes/trainer#transformers.TrainingArguments)
+
+
+## 5. Evaluation
+We provide [a simple script](https://github.com/intel/intel-extension-for-transformers/blob/master/intel_extension_for_transformers/neural_chat/tools/embedding_finetune/evaluate.py) to evaluate the model's performance. We use two metrics: MRR (Mean reciprocal rank) and Hit (Hit Ratio). MRR is an internationally accepted mechanism for evaluating search algorithms. MRR emphasizes the position of ground truth in the retrieval list, the higher it is, the better. Hit emphasizes the accuracy of retrieval, that is, whether the ground truth is included in the retrieval items.
+
+* **Before Finetune**
+```bash
+python evaluate.py \
+--model_name BAAI/bge-base-en-v1.5 \
+--index_file_jsonl_path candidate_context.jsonl \
+--query_file_jsonl_path example.jsonl
+```
+* **After Finetune**
+```bash
+python evaluate.py \
+--model_name BAAI/bge-base-en-v1.5_finetuned \
+--index_file_jsonl_path candidate_context.jsonl \
+--query_file_jsonl_path example.jsonl
+```
+**Some Important Arguments:**
+- `index_file_jsonl_path`: path of JSON data including candidate context where each line is a dict like this:```{"context": List[str]}```.
+- `query_file_jsonl_path`: path of JSON data including queries and positives where each line is a dict like this:```{"query": str, "pos": List[str]}```.
+
+The results should be similar to
```

Comment:
add description for this experiment. the dataset can be selected do not released due to legal concerns. "we conduct a finetuning on an internal business dataset"